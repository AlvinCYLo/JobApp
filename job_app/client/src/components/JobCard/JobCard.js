import React from 'react';
import PropTypes from 'prop-types';

import './JobCard.css';
import TinderCard from 'react-tinder-card';

const JobCard = (props) => {
  const { job } = props;
  const onSwipe = (direction) => {
    console.log(`You swiped: ${direction}`);
  };

  const onCardLeftScreen = (myIdentifier) => {
    console.log(`${myIdentifier} left the screen`);
  };

  return (
    <div className="cardContainer">
      <TinderCard
        className="swipe"
        onSwipe={onSwipe}
        onCardLeftScreen={onCardLeftScreen}
        preventSwipe={['up']}
      >
        <div style={{ backgroundImage: `url(${job.url})` }} className="card">
          <h2>{`${job.company} - ${job.name}`}</h2>
          <h3>{job.keywords}</h3>
        </div>
      </TinderCard>
    </div>
  );
};

JobCard.propTypes = {
  job: PropTypes.shape({
    company: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    keywords: PropTypes.arrayOf(PropTypes.string.isRequired),
  }).isRequired,
};

export default JobCard;
