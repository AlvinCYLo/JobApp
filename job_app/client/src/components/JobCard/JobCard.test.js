import React from 'react';
import renderer from 'react-test-renderer';
import JobCard from './JobCard';

it('renders JobCard correctly', () => {
  const tree = renderer.create(<JobCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
