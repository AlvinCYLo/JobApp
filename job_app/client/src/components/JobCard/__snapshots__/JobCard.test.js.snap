// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders JobCard correctly 1`] = `
<div>
  JOBCARD
</div>
`;
